touchforms/backend/test_xformplayer.py

```diff
@@ -75,7 +75,7 @@ def mock_query(query_function, criteria):
             self.xform,
             session_data=self.session_data,
             form_context={
-                'all_case_ids': ['legolas', 'gimli'],
+                'all_case_ids': ['legolas', 'sauron'],
```

Comment:
this is pretty sketchy - basically as I can tell this line https://github.com/dimagi/commcare/blob/master/cases/src/org/commcare/cases/instance/CaseChildElement.java#L333 always tries to cache something with record ID of 0 when you initialize a form.

the sorting makes it so that the case we are checking (legolas) needs to come alphabetically before any others in the list of IDs.

this is bad, though no worse than the fact that if you flipped the order of `'legolas'` and `'gimli'` in the test on master it would also fail.

@benrudolph fyi